### PR TITLE
`input-number`: `await` wrapped `input-text` to render so it can focus

### PR DESCRIPTION
**Pull request by mostlyfabulous**
[DE43948](https://rally1.rallydev.com/#/29180338367d/custom/486232622040?detail=%2Fdefect%2F601576690591&fdp=true?fdp=true): Update score out of field to be d2l-input-numeric in FACE Assignments

When the `input-number` component is re-rendered (e.g. due to conditional rendering), `autofocus` does not work and component validation isn't triggered correctly for reasons unclear.

Awaiting `input-text` to `updateComplete` let's us access the component when it has finished rendering.

This problem surfaced when trying to update the score input in FACE Assignments to use `input-number` over `input-text`.

### CODE REVIEW

File: components/inputs/input-number.js [hunk 1]
```diff
@@ -371,7 +371,10 @@ class InputNumber extends SkeletonMixin(FormElementMixin(LocalizeCoreElement(Lit
 		});
 	}
 
-	focus() {
+	async focus() {
+		await this.updateComplete;
+		const inputTextElem = this.shadowRoot.querySelector('d2l-input-text');
+		await inputTextElem.updateComplete;
 		const elem = this.shadowRoot.querySelector('d2l-input-text');
 		if (elem) elem.focus();
```
Comment (dlockhart) [inline components/inputs/input-number.js:379]: A pattern we've used elsewhere (like input-text) is this:

```suggestion
		const elem = this.shadowRoot.querySelector('d2l-input-text');
		if (elem) {
			elem.focus();
		} else {
			await this.updateComplete;
			this.focus();
		}
```

Comment (mostlyfabulous) [reply components/inputs/input-number.js:379]: Make sense, will test it out!

**Comment by mostlyfabulous** [comment]
Awesome, it works locally!

**Review by dlockhart — APPROVED** [review]
Looks great, thanks for this!

**Pull request merged**